src/Centreon/Infrastructure/Monitoring/TimelineRepositoryRDB.php

```diff
@@ -0,0 +1,430 @@
+
+        return $timelineEvents;
+    }
+
+    private function isAdmin(): bool
+    {
+        return ($this->contact !== null)
+            ? $this->contact->isAdmin()
+            : false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setContact(ContactInterface $contact): TimelineRepositoryInterface
+    {
+        $this->contact = $contact;
+
+        return $this;
+    }
+
+    /**
+     * @return bool Return FALSE if the contact is an admin or has at least one access group.
+     */
+    private function hasNotEnoughRightsToContinue(): bool
+    {
+        return ($this->contact !== null)
+            ? !($this->contact->isAdmin() || count($this->accessGroups) > 0)
+            : count($this->accessGroups) == 0;
+    }
+
+    /**
+     * Generate SQL query for logs
+     *
+     * @param \Centreon\Infrastructure\CentreonLegacyDB\StatementCollector $collector
+     * @param int $hostId
+     * @param int $serviceId
+     * @return string
+     */
+    protected function generateLogsQuery(StatementCollector $collector, int $hostId, int $serviceId = null): string
+    {
+        $sql = "SELECT
+		CONCAT('L', l.log_id) AS `eventId`,
+        '". LogEventObject::EVENTTYPE ."' AS `eventType`,
+        l.log_id AS `id`,
+		l.output AS `output`,
+		l.ctime AS `timestamp`,
+        l.status AS `status`,
+        l.type AS `type`,
+        l.retry AS `retry`,
+        l.notification_contact AS `contact`,
+        l.notification_cmd AS `command`,
+        NULL AS `persistent`,    
+        NULL as `deletion_time`,
+        NULL AS `start_time`,
+        NULL AS `end_time`,
+        NULL AS `actual_start_time`,
+        NULL AS `actual_end_time`,
+        NULL AS `duration`,
+        NULL AS `started`,
+        NULL AS `cancelled`,
+        NULL AS `fixed`,   
+        NULL AS `sticky`,
+        NULL AS `notify_contacts` 
+        FROM `:dbstg`.`logs` l
+        WHERE l.host_id = :hostId AND l.service_id = :serviceId";
+
+        $collector->addValue(":hostId", $hostId);
+        $collector->addValue(":serviceId", (int) $serviceId);
+
+        // set ACL limitations
+        if (!$this->isAdmin()) {
+            $sql .= " INNER JOIN `:dbstg`.`centreon_acl` AS service_acl ON service_acl.host_id = s.host_id
+                  AND service_acl.service_id = s.service_id
+                  AND service_acl.group_id IN (" . $this->accessGroupIdToString($this->accessGroups) . ")";
+        }
+
+        //Group to avoid duplicate entries
+        $sql .= ' GROUP BY l.log_id';
+
+        return $sql;
+    }
+
+    /**
+     * Generate SQL query for comments
+     *
+     * @param \Centreon\Infrastructure\CentreonLegacyDB\StatementCollector $collector
+     * @param int $hostId
+     * @param int $serviceId
+     * @return string
+     */
+    protected function generateCommentsQuery(StatementCollector $collector, int $hostId, int $serviceId = null): string
+    {
+        $sql = "SELECT
+		CONCAT('C', c.comment_id) AS `eventId`,
```

Comment:
IMHO, eventId should be an integer from only eventId
The unique id will be the association between eventType and eventId

Reply:
this way it wouldn't be unique :) we just needed 1 field that is unique for every timeline set of data

Reply:
We can do a unique identifier based on two fields
For instance, if we want to get information of a specific event, the controller will look like:
`monitoring/events/{type}/{id}`

Reply:
I agree with you, but this 'eventId' property does not need to be used if FE don't need it, and can be removed later if we decide it is useless. For now it does not hurt to have it, because I believe every listing endpoint had to have unique identifier for each row of returned info (due to FE requirements). If that is not the case I can always just omit it from the return of the endpoint. It does not create extra load, it does not mess with any business logic, so in a future refactor we can just remove it easily.